### ISSUE TITLE
Getting "Don't have permission to push. Not trying." message

### ISSUE THREAD

**Issue by kyleniemeyer** [drdoctr/doctr#342]
Although I haven't changed anything except to add some new files, I'm now getting a `Don't have permission to push. Not trying.` message in Travis, which seems to prevent pushing the generated files back to the repo for GitHub Pages to pick up.

Looking at a recent successful build, it looks like that time it did a `git pull` then `git push`, neither of which is happening now.

Any ideas?

**Comment by asmeurer** [comment]
It should print one of these messages indicating why it thinks it shouldn't push: https://github.com/drdoctr/doctr/blob/0f19ff78c8239efcc98d417f36b0a31d9be01ba5/doctr/travis.py#L571-L602

**Comment by asmeurer** [comment]
The exception is if you are using `--no-push` or have `doctr: push: false` in your `.travis.yml`. It won't print a message indicating why in that case.

**Comment by kyleniemeyer** [comment]
OK, I see `The website and docs are not pushed to gh-pages on fork builds`... what does this mean, and why would this be an issue now? (Was this added in a newer version of doctr or something?)

My repo is a fork, but that's because I forked my website template from elsewhere a while ago. How can I override this?

**Comment by asmeurer** [comment]
That's curious if it used to work on the same repo. 

Travis doesn't allow secure environment variables on builds from forked repos (unless things have changed). You may need to delete the repo and recreate it as a non-forked repo.

But again that's curious if it used to work. If that's the case we should figure out what is going on.

**Comment by kyleniemeyer** [comment]
Yeah, it used to work fine, and I haven't changed anything. I hadn't made any changes in a few weeks, that's all.

**Comment by asmeurer** [comment]
OK, I'm guessing the last build was before the most recent release of doctr. The change to not push on forks was new https://drdoctr.github.io/changelog.html, https://github.com/drdoctr/doctr/pull/332. I guess the logic isn't correct, or maybe Travis changed something. I had thought secure environment variables didn't work at all on forks, but I guess that isn't the case. Maybe we should only not push on a fork if the environment variable isn't there.

**Comment by asmeurer** [comment]
If you need a workaround you can manually install doctr 1.7.4.

**Comment by asmeurer** [comment]
Fix started at https://github.com/drdoctr/doctr/pull/343. I still need to fix the tests and do some manual testing. If you can check if that branch works on your repo that would help.

**Comment by kyleniemeyer** [comment]
Can I get Travis to use that one somehow, or should I just try locally?

**Comment by asmeurer** [comment]
It won't work locally. You can install it on Travis using `pip install git+git://github.com/drdoctr/doctr@forks`.

**Comment by kyleniemeyer** [comment]
@asmeurer thanks—I actually figured that out after commenting here... build is in progress.

So, I tried using that fork, but what's happening is weird... I see this message `The last commit on this branch was pushed by doctr. Not pushing to avoid an infinite build-loop.`, even after pushing a few content changes.

The few log is at https://gist.github.com/kyleniemeyer/0da4ea0d3f175c7401ca9f5f53a4dd31

**Comment by kyleniemeyer** [comment]
Though it does look like it's successfully pushing the generated content. No idea.

**Comment by asmeurer** [comment]
That happens if you run Travis on the GitHub pages branch itself (`gh-pages` for most repos, `master` for `*.github.io` repos). That's because doctr pushes up to that branch, so otherwise it would get caught in an infinite build loop.

**Comment by kyleniemeyer** [comment]
I didn’t think I was- I use the “source” branch for the source files. But maybe I don’t have it configured completely right.

**Comment by banesullivan** [comment]
Is this close to being resolved? I've been using `doctr` for a number of my open-source projects and I love it! 

I have a side project that uses several open-source projects I develop and thought it'd be cool to showcase the project in a `sphinx-gallery`. I set up the docs to build and they were deploying from Travis via `doctr` but then I filled out the rest of the project and it requires more resources than Travis can provide (13 GB of ram at least and preferably a GPU) - I was thinking that I'd manually build and deploy the docs from my local machine since I won't edit the project moving forward but I'm getting this issue when I run:

```bash
doctr deploy --force --built-docs ./docs/_build/html .
```

The project is: https://github.com/OpenGeoVis/GeothermalDesignChallenge

**Comment by asmeurer** [comment]
I do intend to fix this issue. It requires some manual testing to get any fix like this, which is why I've been a bit slow. I'll try to get to it this week.

However, this sounds like a separate issue from what you are having. doctr was never designed to be run locally. We could probably make it work, but it would require some changes.

**Comment by banesullivan** [comment]
> However, this sounds like a separate issue from what you are having. doctr was never designed to be run locally. We could probably make it work, but it would require some changes.

How big of changes?

**Comment by asmeurer** [comment]
I've honestly never thought of doing it, so I'm not sure. I'd have to take a look to see. Feel free to open an issue for it.

**Comment by asmeurer** [comment]
I opened https://github.com/drdoctr/doctr/issues/345 about it.
